docs/dev/generator.md

```diff
@@ -0,0 +1,201 @@
+# Importer Documentation
+
+This document details the importer process.
+
+## Overview
+
+The implementation is divided into two components: generic and NetBox specific.
+
+### Generic Component
+
+The generic component encompasses the `NautobotAdapter` and `SourceAdapter` capable of generating `DiffSyncModel` classes based on the input data, the current version of Nautobot, and any deviations defined in the source configuration.
+
+It comprises the following modules, located in the `nautobot_netbox_importer/generator` directory:
+
+- `base.py`: Implements base classes and utilities utilized throughout the application, also establishing constants and shared type definitions.
+- `nautobot.py`: Provides the `NautobotAdapter`, along with `NautobotModelWrapper` and `NautobotFieldWrapper` classes that define the Nautobot structure.
+- `source.py`: Offers the `SourceAdapater`, `SourceModelWrapper` and `SourceField` classes that define the source structure.
+- `fields.py`: Contains factories for fields deviation definitions.
+- `summary.py`: Supplies utility functions for summarizing data structures and presenting import statistics.
+
+### NetBox Specific Component
+
+The NetBox specific component is further segmented into sections, located in the `nautobot_netbox_importer/diffsync` directory:
+
+- `adapters/nautobot.py`: Inherits `NautobotAdapter` from the generic `NautobotAdapter` and implements pieces necessary for SSoT job.
```

Comment:
why do we need a generic NautobotAdapter to create yet another NautobotAdapter? what is this abstraction providing?

Reply:
The first one is DiffSync only related implementation. The other one is adding SSoT related things, but it's not probably necessary.

The reason to create the second one was to make the file structure the same way as is in other SSoT apps / integrations.